docs/notebooks/documents/getting_started.py

```diff
@@ -0,0 +1,136 @@
+# object. Specifically, the extension defines a magic keyword, `skip`, that takes a value or
+# expression that, if it evaluates to True, causes the code section to be skipped during notebook
+# execution.
+#
+# The next section in the notebook configures information about the contract object associated with
+# the notebook. In some cases, the variables will set by the notebook using the Papermill
+# extensions. In some cases, the variables may be customized for the specific behavior desired.
+#
+# The following section initializes the PDO environment from the PDO configuration files. There may
+# be opportunities for customization, but so long as the PDO configuration is complete changes to
+# this section should be infrequent.
+#
+# The final common section initializes the contract context. Where the PDO initialization handles
+# configuration of the PDO modules, this section handles configuration of the specific contract
+# object and its relationship to other contract objects.
+#
+# %% [markdown]
+# ## Initialize the Environment
+#
+# %%
+import os
+import pdo.contracts.jupyter as pc_jupyter
+import IPython.display as ip_display
+import ipywidgets
+
+pc_jupyter.load_ipython_extension(get_ipython())
+
+try : state
+except NameError:
+    (state, bindings) = pc_jupyter.initialize_environment('unknown')
+
+# %% [markdown]
+# ## Create Keys
+#
+# PDO uses ECDSA cryptographic keys to sign transactions. Each key pair represents an identity or
+# role. The list below will show a list of currently available keys. There is a form to use to
+# create new keys (the refresh button will update the key list when you create new keys).
+# Additional management of keys is provided by the [Key Manager Notebook](key_manager.ipynb).
+#
+# %%
+from pdo.contracts.keys import PrivateKeyListWidget, GenerateKeyWidget
+
+key_list = PrivateKeyListWidget(state, bindings)
+key_gen = GenerateKeyWidget(state, bindings)
+ip_display.display(ipywidgets.VBox([key_list, key_gen]))
+
+# %% [markdown]
+# ## Import Site Information
+#
+# PDO clients require information about the services that will be used for contracts. Typically, a
+# service provider will create a site description file (often called `site.toml`) that can be used
+# to import information about a number of services.
+#
+# The widget below contains information about the currently created service groups and a form
+# for uploading a new site file. Additional management of service information is provided by the
+# [Service Manager Notebook](service_manager.ipynb)
+#
+# %%
+from pdo.contracts.services import ServiceListWidget
+from pdo.contracts.services import ServiceUploadWidget
+from pdo.contracts.services import service_labels
+
+children = map(lambda stype : ServiceListWidget(state, bindings, stype), service_labels.keys())
+service_list = ipywidgets.Tab(children=list(children), titles=list(service_labels.values()))
+ip_display.display(ipywidgets.VBox([service_list, ServiceUploadWidget(state, bindings)]))
```

Comment:
looks like you assume that by default there  are services deployed on localhost? What if there all services are located on remote only?  I see that there is an option to upload the site.toml file, but the option is presented only after a default setting on localhost is executed.

Reply:
where do you see localhost? there are no assumptions. yes, test_jupyter sets up a localhost site (that's where the services are located). but you can load any site file. i do this often for other "permanent" services.

Reply:
ok, when I ran the notebook, my observation was the I get the option to upload the site.toml only after executing the cell; but executing the cell meant that it automatically showed the existence of a default localhost based group. My question was what happens if such a default localhost group does not exist. I have no way to test it now, until I deploy a more elaborate set up.

Reply:
yes. we had already loaded that site.toml for the localhost test files.